lib/grpcx/server/interceptor/active_record/errors.rb

```diff
@@ -0,0 +1,39 @@
+module Grpcx
+  module Server
+    module Interceptor
+      module ActiveRecord
+        # Intercepts most common ActiveRecord::ActiveRecordError-s
+        # and transforms them into corresponding GRPC::BadStatus ones.
+        class Errors < GRPC::ServerInterceptor
```

Comment:
I think this should not be activerecord specific, but a general Interceptor::Rescue, where you can register additional error classes, see e.g. https://github.com/rails/rails/blob/375a4143cf5caeb6159b338be824903edfd62836/actionpack/lib/action_controller/metal/rescue.rb and https://github.com/rails/rails/blob/375a4143cf5caeb6159b338be824903edfd62836/activesupport/lib/active_support/rescuable.rb